rfcs/0046-platform-support-tiers.md

```diff
@@ -0,0 +1,431 @@
+---
+feature: platform_support_tiers
+start-date: 2019-04-28
+author: Michael Raskin
+shepherd-team: Ryan Mulligan, Jonas Pfenniger, Graham Christensen, John Ericson
+shepherd-leader: John Ericson
+co-authors: Matthew Bauer
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Document the way to describe the level of support a platform can expect in
+Nixpkgs.
+
+# Motivation
+[motivation]: #motivation
+
+When adding a new platform, be it a new C library option, a new
+cross-compilation target configuration or a new CPU architecture, there is a
+discussion of support expectations and maintenance burden. Having a documented
+vocabulary to describe the expectations and documented precedents should make
+such discussions more efficient. Building a consensus about a standard list of
+restrictions for less-popular platform support can also streamline most
+of the decisions of this type.
+
+# Detailed design
+[design]: #detailed-design
+
+## Concrete actions
+
+If this RFC is accepted, the following changes will happen:
+
+* The definition of support tiers and a list of platforms with the
+corresponding support tiers is added as an appendix to the Nixpkgs manual. The
+list of platforms is further maintained as a part of the Nixpkgs manual. The
+platform list in the manual will be initially based on the list in the appendix
+of the present RFC.
+
+* Other official documentation and marketing materials of the Nix/Nixpkgs/NixOS
+project are edited and maintained to be consistent with the platform list in
+the manual.
+
+* NixOS GitHub organisation teams @NixOS/aarch64-maintainers and
+@NixOS/exotic-platform-maintainers are created.
+
+* Platform support status at the Tier 4 and below is updated according to
+achievement of the requirements defined in this RFC.
+
+* Discussions of platform support at the Tier 3 and above start using the
+vocubulary based on this RFC and ensure that the requirements of this RFC are
+met.
+
+## Platform elements
+
+* CPU architecture (possibly: subarchitecture, optional features)
+* OS ABI
+* C compiler
+* C library
+* OS kernel
+* NixOS/non-NixOS global layout, in case of Linux with glibc
+* Global linking options
+
+## Questions to answer for each platform
+
+* What fraction of packages is expected to work?
```

Comment:
```suggestion
* What fraction of packages are expected to work?
```

Reply:
(looking up) oh indeed, the modern majority position is «are», even if there are still some style guides that favour «is» here (of course I only searched five minutes ago because when writing it was obviously «a fraction is» to me).